### ISSUE TITLE
Wrong SQL query on document data deletes the document data

### ISSUE THREAD

**Issue by Anphisa** [polypheny/Polypheny-DB#437]
### Polypheny-DB Version

339fd10f9dc23f7c8daf8affbe6f9e01d11ad8eb

### Affected Areas

DB, UI

### Browser

Chrome

### Client Operating System

Windows 10

### Bug Description

When I use a wrong SQL query on document data, I seem to delete the document data.

### Reproduction Steps

- Add a document namespace
- Put document data in the namespace. I used this MQL code:
```
db.UserProperties.insert( { userid: "0", age: 20, eyecolor: "brown" } )
db.UserProperties.insert( { userid: "1", age: 40 } )
```
- Query the document namespace with a SQL expression that is wrong. I used this query:
```
SELECT * FROM gavel_doc."userid"
```

This throws an error:

Error:
From line 1, column 15 to line 1, column 32: Object 'userid' not found within 'gavel_doc'

But it also deletes the document data that was added with the MQL code above. At least, it isn't visible via the UI "Data" element anymore.

### Expected Behavior

Throw the same error, but keep the data.

**Comment by datomo** [comment]
Hey @Anphisa, thank you for this issue. 
Due to a missing commit on the catalog the implicit creation of the collection with the `db.UserProperties.insert( { userid: "0", age: 20, eyecolor: "brown" } )` statement was still not fixed. This was then wrongfully also part of the rollback due to the incorrect query after. 
This should be fixed in #437 but requires some testing as I have moved the commit of the catalog into the commit of the transaction itself. 

To temporarily bypass this error from happening, the explicit creation of the collection can be used:
`db.createCollection(("UserProperties")`
 This has to be executed between the creation of the namespace and the insertion.

**Comment by datomo** [comment]
Fixed in #438.
